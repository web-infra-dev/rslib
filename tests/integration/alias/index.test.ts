import { buildAndGetResults } from 'test-helper';
import { expect, test } from 'vitest';

test('source.alias', async () => {
  const fixturePath = __dirname;
  const { entries } = await buildAndGetResults(fixturePath);

  expect(entries.esm).toContain('hello world');
  expect(entries.cjs).toContain('hello world');

  // simple artifacts check
  expect(entries.esm).toMatchSnapshot();
  expect(entries.cjs).toMatchSnapshot();
});
