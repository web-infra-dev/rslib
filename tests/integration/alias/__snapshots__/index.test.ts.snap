// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`source.alias 1`] = `
"const a = 'hello world';
console.info(a);
"
`;

exports[`source.alias 2`] = `
""use strict";
var __webpack_exports__ = {};
const a = 'hello world';
console.info(a);
var __webpack_export_target__ = exports;
for(var i in __webpack_exports__)__webpack_export_target__[i] = __webpack_exports__[i];
if (__webpack_exports__.__esModule) Object.defineProperty(__webpack_export_target__, '__esModule', {
    value: true
});
"
`;
